### PR TITLE
feat: Add plugin IDs array support for agents

### PR DESCRIPTION
**Pull request by OAGr**
## Summary
- Add `pluginIds` field to AgentVersion schema and TypeScript types
- Update `analyzeDocument` to use plugin-based routing with `analyzeDocumentUnified`
- Maintain backward compatibility with `extendedCapabilityId`
- Add comprehensive test coverage for plugin routing and legacy fallback

## Key Changes
- **Database**: Added `pluginIds String[]` field to `AgentVersion` model
- **Backend Logic**: Modified `analyzeDocument.ts` to route based on `pluginIds` array
- **Type Safety**: Updated Zod schemas to include `pluginIds` with `PluginType` enum
- **Backward Compatibility**: `pluginIds` takes precedence over `extendedCapabilityId` when both present
- **Testing**: 5 comprehensive test cases covering plugin routing and legacy fallback

## Behavior
- When `pluginIds` specified: Uses `analyzeDocumentUnified` with selected plugins
- When `pluginIds` empty/undefined: Falls back to legacy `extendedCapabilityId` logic
- Prioritization: `pluginIds` overrides `extendedCapabilityId` when both present

## Test Results
✅ All 5 plugin routing tests pass
✅ TypeScript compilation passes for @roast/ai and @roast/web  
✅ Backward compatibility maintained

This replaces the hardcoded workflow selection in `analyzeDocument.ts` with a flexible plugin-based system that uses the `analyzeDocumentUnified` workflow.

🤖 Generated with [Claude Code](https://claude.ai/code)

___

### **PR Type**
Enhancement

___

### **Description**
- Add `pluginIds` array field to AgentVersion schema for flexible plugin selection

- Update `analyzeDocument` to route via `analyzeDocumentUnified` when plugins specified

- Maintain backward compatibility with existing `extendedCapabilityId` logic

- Add comprehensive test coverage for plugin routing scenarios

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Agent with pluginIds"] --> B["analyzeDocumentUnified"]
  C["Agent with extendedCapabilityId"] --> D["Legacy workflows"]
  E["Agent with both"] --> B
  B --> F["Plugin-based analysis"]
  D --> G["Capability-based analysis"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>agentSchema.ts</strong><dd><code>Add pluginIds field to agent schemas</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal-packages/ai/src/types/agentSchema.ts

<ul><li>Import <code>PluginType</code> enum for type validation<br> <li> Add optional <code>pluginIds</code> field to <code>AgentVersionSchema</code> and <code>BaseAgentSchema</code><br> <li> Use <code>z.array(z.nativeEnum(PluginType))</code> for type-safe plugin array <br>validation</ul>

</details>

  </td>
  <td><a href="https://github.com/quantified-uncertainty/roast-my-post/pull/171/files#diff-7e4a9d869251a76ff8b18b7ade89202349f506eedbcbfe60180eb4047a5912fc">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>analyzeDocument.ts</strong><dd><code>Implement plugin-based workflow routing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal-packages/ai/src/workflows/documentAnalysis/analyzeDocument.ts

<ul><li>Import <code>analyzeDocumentUnified</code> and <code>PluginType</code> for plugin-based routing<br> <li> Add plugin routing logic that takes precedence over <br><code>extendedCapabilityId</code><br> <li> Route to <code>analyzeDocumentUnified</code> when <code>pluginIds</code> array is populated<br> <li> Update logging to distinguish between plugin-based and legacy <br>workflows</ul>

</details>

  </td>
  <td><a href="https://github.com/quantified-uncertainty/roast-my-post/pull/171/files#diff-09cbe43670b1fb2b3ace967422b4b5b22b8200f67f087c9cc7108485b845aad1">+18/-4</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>schema.prisma</strong><dd><code>Add pluginIds database field</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal-packages/db/prisma/schema.prisma

<ul><li>Add <code>pluginIds String[]</code> field to <code>AgentVersion</code> model<br> <li> Set default value to empty array for new records</ul>

</details>

  </td>
  <td><a href="https://github.com/quantified-uncertainty/roast-my-post/pull/171/files#diff-2f79b7f4d8cb61c0e2832d98f50a5d46f3b6c019227e0232d72bfae412dc9863">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>analyzeDocument.plugin-ids.test.ts</strong><dd><code>Add plugin routing test coverage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal-packages/ai/src/workflows/documentAnalysis/__tests__/analyzeDocument.plugin-ids.test.ts

<ul><li>Create comprehensive test suite for plugin-based routing functionality<br> <li> Mock <code>analyzeDocumentUnified</code> and legacy workflows for isolated testing<br> <li> Test plugin routing, backward compatibility, and priority handling<br> <li> Cover edge cases like empty arrays and mixed configurations</ul>

</details>

  </td>
  <td><a href="https://github.com/quantified-uncertainty/roast-my-post/pull/171/files#diff-57dc257fc584067e8ffda164b18e805eb73dc91cbc331dacd86f572b03717e13">+164/-0</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by OAGr** [comment]
/review focus on the last 2 commits

File: test-spelling-e2e.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env node
```
Comment (OAGr) [inline test-spelling-e2e.js:1]: remove this file

File: internal-packages/ai/REFACTORING_NOTES.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
+# Plugin System Refactoring Notes
```
Comment (OAGr) [inline internal-packages/ai/REFACTORING_NOTES.md:1]: This file needs to be a removed

File: internal-packages/ai/jest.config.js [hunk 3]
```diff
@@ -22,5 +29,15 @@ module.exports = {
...
   ],
-  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts']
+  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
+  // Handle ES modules properly
+  transformIgnorePatterns: [
+    'node_modules/(?!(nanoid|@roast)/)'
+  ],
+  extensionsToTreatAsEsm: ['.ts'],
+  moduleNameMapper: {
+    '^nanoid$': '<rootDir>/../../node_modules/.pnpm/nanoid@3.3.11/node_modules/nanoid/index.cjs'
```
Comment (OAGr) [inline internal-packages/ai/jest.config.js:39]: Why call out nanoid? Maybe nanoid can be replaced, if its a pain? This also seems like it'll break the moment and we upgrade nanoid

File: internal-packages/ai/src/index.ts [hunk 4]
```diff
@@ -114,7 +122,7 @@ export type {
 } from './shared/types';
 
 export {
-  DEFAULT_TIMEOUT,
+  DEFAULT_TIMEOUT, // Keep for backward compatibility
```
Comment (OAGr) [inline internal-packages/ai/src/index.ts:125]: Does this imply we have dead code? If so, we should refactor.

File: internal-packages/ai/src/workflows/documentAnalysis/shared/llmUtils.ts [hunk 5]
```diff
@@ -16,8 +16,8 @@ interface LogDetails {
       output_tokens: number;
     };
   };
-  context: Record<string, any>;
-  outputStats: Record<string, any>;
+  context: Record<string, unknown>;
```
Comment (OAGr) [inline internal-packages/ai/src/workflows/documentAnalysis/shared/llmUtils.ts:19]: Can we do better than 'unknown'?
